### DISCUSSION TITLE
How to populate Tables/Screens with the external Data Sources' data?

### DISCUSSION THREAD

**Discussion by arladmin** [Budibase/budibase#1032] [Help]
I've added an external data source, currently Airtable.

I would like to know if/how we can use the saved queries for these external data sources to design Tables and subsequently Screens, automatically? (Sorry if this is already an option and i can't see it.)

**Basically**: how to use the external data sources?

Thanks.

**Comment by mjashanks** [comment]
Working on an answer for you...

**Comment by mjashanks** [comment]
Hey @arladmin .

Currently, the auto-screens feature does not work with external data sources. We will look to add this.

You can create a screen from scratch then add a grid component, as shown here: https://www.loom.com/share/9db51a82b99d42b496c5eef57d79b8f2

If you really, really want to use autoscreens, you can follow this workaround: https://www.loom.com/share/3718695d730f447294d9f677f97bc4b8

1. Create auto screen for "Users".
2. Change the name & URL in the setup screen
3. Edit the data source on the grid, to point to your Airtable query.

The issue with this, is that you will now have some broken links on that page:

- "Create New" button
- "View" button, on each row

You can fix these links, using the steps below. 

1. Edit the Detail URL on the grid to `/your-update-query-name/:id`
2. Edit the "Create New" button's "Define Actions", to point to `/your-create-query-name/new/row`

Of course, you then need to create screens for these new URLs! Let me know if that is also something that you need help with :)
